TheMET/TheMET/Network/NetworkManager.swift

```diff
@@ -10,38 +10,40 @@ import Foundation
 class NetworkManager {
     
     func get(urlString: String, parameters: [String : String], completion: @escaping ([String: Any]?) -> Void)  {
-        guard var components = URLComponents(string: urlString) else {
-            completion(nil)
-            return
-        }
-        components.queryItems = parameters.map { (key, value) in
-            URLQueryItem(name: key, value: value)
-        }
-        components.percentEncodedQuery = components.percentEncodedQuery?.replacingOccurrences(of: "+", with: "%2B")
-        guard let url = components.url else {
-            completion(nil)
-            return
-        }
-        var urlRequest = URLRequest(url: url)
-        urlRequest.httpMethod = "GET"
-        let urlSession = URLSession.shared
-        let fileDownloadTask = urlSession.dataTask(
-            with: urlRequest
-        ) { data, _, error in
-            let executeCompletionOnMain: ([String: Any]?) -> Void = { loadedData in
-                DispatchQueue.main.async {
-                    completion(loadedData)
-                }
+        DispatchQueue.global().asyncAfter(deadline: DispatchTime.now() + .milliseconds(Int(12.5)), qos: .background) {
```

Comment:
This code does not limit the amount of network request to 80 per second. It just postpones every single request by 12.5ms. For example, you can request 1000 requests at the same time and they just will be delayed by 12.5ms and then requested all together.

How to limit the amount of requests.
- You need to record the time when each request happens to some array
- Every time a new request is about to happen you need to update this array by removing outdated timings (more than 1 sec in the past)
- Then you need to check the count of the rest timings in array. If it is less than 80, you can proceed and do the actual network request. If it is more or equal to 80, the whole procedure needs to be done again later (The precise delay can be calculated out of the timings array or you can use constant like 12.5ms)
- The the actual request is happening you need to put the act time of it to the array of timings.